FreeRTOS/FreeRTOS-Plus-TCP
# [PR#482] - Update ARP cache from gratuitous ARP packets v2

This PR updates [PR#482](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/482) with respect to latest main branch changes.

Description
-----------
_[Original PR description]_
The changes proposed in this PR were presented earlier in PR #463. Now I have adapted it to the new source structure of the library that was introduced about 3 weeks ago.

When receiving a gratuitous ARP **request**, and if there is a cache entry for that protocol address, the cache entry should be updated with the MAC-address provided,

The "issue" was reported on the FreeRTOS forum as [FreeRTOS+TCP not updating ARP table with gratuitous ARP request with a changed MAC address
Libraries](https://forums.freertos.org/t/freertos-tcp-not-updating-arp-table-with-gratuitous-arp-request-with-a-changed-mac-address/14670).
Thank you [@nlangenb](https://forums.freertos.org/u/nlangenb) for this.

Test Steps
-----------
Ran demo application on STM32Hxx

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
